cbauthx/authcheck_http.go

```diff
@@ -80,6 +81,17 @@ func (a *AuthCheckHttp) CheckUserPass(ctx context.Context, username string, pass
 		}
 	}
 
+	// In the interest of mitigating the impact of an incorrect auth check url
+	// being used, we validate some expected state of the response which has the
+	// effect of significantly reducing the chances we accept auth from those
+	// urls which do not actually check auth.
```

Comment:
What is the impact of using an incorrect auth check URL that is being mitigated here?

Reply:
Imagine if for some reason, this was initialized with an authCheckURL of "", when we concatenate the source server url of "http://localhost:8091" to this, we would accidentally be using the index-page for Couchbase server rather than the _cbauth url.  In this case, by checking that the response properties match our expectation, it ensures we are unlikely to accept a pages response unless it was a cbauth response.

Reply:
I see, thanks for the exaplination